implementations/rust/ockam/ockam_identity/src/secure_channel/handshake/handshake_worker.rs

```diff
@@ -368,7 +368,17 @@ impl HandshakeWorker {
 
         // create a separate encryptor worker which will be started independently
         {
-            let rekeying = !self.key_exchange_only;
+            let rekeying;
```

Comment:
This could be:
```rust
let (rekeying, credential_retriever) = 
   if self.key_exchange_only {
      (false, None)
   } else {
      (true, self.credential_retriever.clone())
   }
```